src/jit/codegenxarch.cpp

```diff
@@ -5441,13 +5499,19 @@ void CodeGen::genConsumePutStructArgStk(
 
     // If the op1 is already in the dstReg - nothing to do.
     // Otherwise load the op1 (GT_ADDR) into the dstReg to copy the struct on the stack by value.
+#ifdef _TARGET_X86_
+    assert(dstReg != REG_SPBASE);
+    inst_RV_RV(INS_mov, dstReg, REG_SPBASE);
+#else // !_TARGET_X86_
     if (dstAddr->gtRegNum != dstReg)
     {
         // Generate LEA instruction to load the stack of the outgoing var + SlotNum offset (or the incoming arg area
         // for tail calls) in RDI.
         // Destination is always local (on the stack) - use EA_PTRSIZE.
-        getEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, dstReg, baseVarNum, putArgNode->getArgOffset());
+        getEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, dstReg, m_stkArgVarNum, putArgNode->getArgOffset());
+        assert(m_stkArgVarNum != BAD_VAR_NUM);
```

Comment:
Nit: it seems to make more sense to have the assert before the preceding call rather than after it.